plugins/inventory/insights.py

```diff
@@ -243,11 +244,11 @@ def parse(self, inventory, loader, path, cache=True):
                 per_page = response.json()['per_page']
                 page = response.json()['page']
                 if per_page * (page - 1) + count < total:
-                    url = "%s&page=%s" % (url, (page + 1))
+                    hosts_url = "%s&page=%s" % (url, (page + 1))
                     if len(filter_tags) > 0:
-                        url = "%s&tags=%s" % (url, '&tags='.join(filter_tags))
+                        hosts_url = "%s&tags=%s" % (url, '&tags='.join(filter_tags))
```

Comment:
Actually, this one should probably be:
```suggestion
                        hosts_url = "%s&tags=%s" % (hosts_url, '&tags='.join(filter_tags))
```
Otherwise, the `&page=` params above would get entirely overwritten.